WooCommerce/Classes/ViewRelated/Products/Edit Product/Downloads/Downloadable FIle List/ProductDownloadDnD.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import CoreServices
+import Yosemite
+
+/// A wrapper around ProductDownload, to make it compatible in using as Drag and Drop data source in an table view. Represents a ProductDownload entity.
+///
+public class ProductDownloadDnD: NSObject, Codable {
+    public let download: ProductDownload
+
+    /// ProductDownloadDnD initializer.
+    ///
+    public init(download: ProductDownload) {
+        self.download = download
+    }
+
+    /// Public initializer for ProductDownloadDnD
+    ///
+    required public convenience init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let download = try container.decode(ProductDownload.self, forKey: .download)
+        self.init(download: download)
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(download, forKey: .download)
+    }
+}
+
+extension ProductDownloadDnD: NSItemProviderReading, NSItemProviderWriting {
+    public static var readableTypeIdentifiersForItemProvider: [String] {
+        return [(kUTTypeUTF8PlainText) as String]
+    }
```

Comment:
I'll suggest you also to add a comment about it. It's not totally clear the purpose of this. 👍 
```suggestion
    static var readableTypeIdentifiersForItemProvider: [String] {
        return [(kUTTypeUTF8PlainText) as String]
    }
```

Reply:
RIght. Added the comment.